src/Validation/Requires.cs

```diff
@@ -469,21 +469,26 @@ public static void That([DoesNotReturnIf(false)] bool condition, string? paramet
         /// <exception cref="ArgumentException">Thrown if the tested condition is false.</exception>
         /// <exception cref="ArgumentNullException">Thrown if <paramref name="predicate"/> is <see langword="null"/>.</exception>
         [DebuggerStepThrough]
-        public static void ValidElements<T>([ValidatedNotNull] IEnumerable<T>? values, Predicate<T> predicate, string parameterName, string message)
+        public static void ValidElements<T>([ValidatedNotNull] IEnumerable<T> values, Predicate<T> predicate, string parameterName, string message)
         {
+            // To whoever is doing random code cleaning:
+            // Consider the performance when changing the code to delegate to NotNull.
+            // In general do not chain call to another function, check first and return as earlier as possible.
```

Comment:
Do you have data to back this up? The JIT is really good at inlining small method calls. It doesn't often inline a method that has a throw statement in it, however, as `NotNull` does. So you might be right, but it probably bears measuring. Anyway, you can leave this here and the comment. I just thought I'd call this out in case you had data, I'd be interested to hear it.
nit: earlier -> early

Reply:
LOL. This was a copy/paste from all the other methods that have it. I'll fix it, and then remove them in all of the methods in favor of calling NotNull based on #45.